### PR TITLE
Update interleaved index and count(*) known limitations

### LINKED ISSUE #1865: sql: support star expansion in sub-expressions
PR: https://github.com/cockroachdb/cockroach/pull/17833

Resolves this [known limitation](https://www.cockroachlabs.com/docs/stable/known-limitations.html#counting-distinct-rows-in-a-table).

### PR DESCRIPTION
**Pull request by jseldess**
Based on https://github.com/cockroachdb/cockroach/pull/17860 and https://github.com/cockroachdb/cockroach/pull/17833

- Remove from 1.1 known limitations.
- In 1.0 known limitations, indicate they've been resolved in 1.1.
- In 1.0 known limitations, make "resolved" notes more visible.

Fixes #1865

### CODE REVIEW

**Comment by jseldess** [comment]
@m-schneider, PTAL.

**Comment by m-schneider** [comment]
:lgtm:

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1864#-:-KsdDgmuW9VeqBqKBBRv:bnfp4nl)*

File: v1.0/known-limitations.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ This page describes limitations we've identified in the [CockroachDB 1.0](../rel
 
 ## Removing all rows from large tables
 
+{{site.data.alerts.callout_info}}Resolved as of version 1.1. See <a href="https://github.com/cockroachdb/cockroach/pull/17016">#17016</a>.{{site.data.alerts.end}}
```
Comment (knz) [inline v1.0/known-limitations.md]: It's only resolved for TRUNCATE. The problem persists for DELETE.

Comment (jseldess) [reply v1.0/known-limitations.md]: Ugh. I didn't realize that. @vivekmenezes, are we likely to resolve this limitation on `DELETE` as well for 1.1?

Comment (knz) [reply v1.0/known-limitations.md]: FYI it's tracked in https://github.com/cockroachdb/cockroach/issues/17921

**Pull request merged**